cognite/client/data_classes/data_modeling/filters.py

```diff
@@ -0,0 +1,312 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Optional, Sequence, Tuple, Union, cast, final
+
+ContainerReference = Tuple[str, str]
+ViewReference = Union[Tuple[str, str], Tuple[str, str, str]]
```

Comment:
Let's use ContainerIdentifier and ViewIdentifier here instead.

Reply:
That will require also handling the dumping/loading of ContainerId/ViewId classes in HasData filters (along with tests for cases where those classes are being used).